File: public/public-web/src/test/java/ch/difty/scipamato/publ/web/font/MetaOTFontResourceProviderTest.java
package ch.difty.scipamato.publ.web.font;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import ch.difty.scipamato.publ.config.ApplicationPublicProperties;
import ch.difty.scipamato.publ.web.resources.MetaOTCssResourceReference;

@ExtendWith(MockitoExtension.class)
class MetaOTFontResourceProviderTest {

    private MetaOTFontResourceProvider provider;

    @Mock
    private ApplicationPublicProperties applicationProperties;

    @Test
    void withNoCommercialFontPresentSetting_getsNull() {
        when(applicationProperties.isCommercialFontPresent()).thenReturn(false);

        provider = new MetaOTFontResourceProvider(applicationProperties);
        assertThat(provider.getCssResourceReference()).isNull();
        assertThat(provider.isCommercialFontPresent()).isFalse();

        verify(applicationProperties).isCommercialFontPresent();
    }

    @Test
    void withCommercialFontPresentSetting_getsReference() {
        when(applicationProperties.isCommercialFontPresent()).thenReturn(true);

        provider = new MetaOTFontResourceProvider(applicationProperties);
        assertThat(provider.getCssResourceReference())
            .isNotNull()
            .isInstanceOf(MetaOTCssResourceReference.class);
        assertThat(provider.isCommercialFontPresent()).isTrue();

        verify(applicationProperties).isCommercialFontPresent();
    }
}

File: public/public-web/src/test/kotlin/ch/difty/scipamato/publ/web/font/MetaOTFontResourceProviderTest.kt
package ch.difty.scipamato.publ.web.font

import ch.difty.scipamato.publ.config.ApplicationPublicProperties
import ch.difty.scipamato.publ.web.resources.MetaOTCssResourceReference
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.amshove.kluent.shouldBeFalse
import org.amshove.kluent.shouldBeInstanceOf
import org.amshove.kluent.shouldBeNull
import org.amshove.kluent.shouldBeTrue
import org.junit.jupiter.api.Test

internal class MetaOTFontResourceProviderTest {

    @Test
    fun withNoCommercialFontPresentSetting_getsNull() {
        val applicationProperties = mockk<ApplicationPublicProperties> {
            every { isCommercialFontPresent } returns false
        }
        with(MetaOTFontResourceProvider(applicationProperties)) {
            cssResourceReference.shouldBeNull()
            isCommercialFontPresent.shouldBeFalse()
        }
        verify { applicationProperties.isCommercialFontPresent }
    }

    @Test
    fun withCommercialFontPresentSetting_getsReference() {
        val applicationProperties = mockk<ApplicationPublicProperties> {
            every { isCommercialFontPresent } returns true
        }
        with(MetaOTFontResourceProvider(applicationProperties)) {
            cssResourceReference shouldBeInstanceOf MetaOTCssResourceReference::class
            isCommercialFontPresent.shouldBeTrue()
        }
        verify { applicationProperties.isCommercialFontPresent }
    }
}
